pennylane/variable.py

```diff
@@ -94,12 +92,12 @@ class VariableRef:
     #: dict[str->array[float]]: current auxiliary parameter values, set in :meth:`.BaseQNode._set_variables`
     kwarg_values = None
 
-    def __init__(self, idx, name=None, is_kwarg=False):
+    def __init__(self, idx, name=None, basename=None):
         self.idx = idx  #: int: parameter index
         self.name = name  #: str: parameter name
-        self.idx = idx  #: int: parameter index
+        self.basename = basename  #: str: for auxiliary parameters the key for the kwarg_values dict
+        self.is_kwarg = basename is not None
```

Comment:
What is the reason for storing this in two attributes?

Reply:
it could be a property.